internal/pkg/config/config.go

```diff
@@ -137,7 +161,39 @@ func (vacuousMatcher) MatchString(string) bool {
 	return true
 }
 
-// Returns the first non-nil matcher.  If all are nil, returns a vacuousMatcher.
+type fieldTagMatcher struct {
+	Key string
+	Val string
+}
+
+type rawFieldTagMatcher struct {
+	Key string
+	Val string
+}
+
+func (ft *fieldTagMatcher) UnmarshalJSON(bytes []byte) error {
+	// Unknown config fields are not allowed.
```

Comment:
Since different matchers use different stringsets for validation, the only common part is "unmarshal to string map". That's not a lot of duplication to remove.

Reply:
I was thinking of something like:
```go
func validateFields(bytes []byte, validFields map[string]bool, matcherName string) error {
    rawMap := make(map[string]interface{})
	if err := json.Unmarshal(bytes, &rawMap); err != nil {
		return err
	}
	for label := range rawMap {
		if !validFields[label] {
			return fmt.Errorf("%q is not a valid config field for %q, expect one of: %v", label, matcherName, validFieldTagMatcherFields)
   }
   return nil
}
```
If we need to change this logic at some point, or e.g. the error message, we'll be able to change it in only one place, instead of three.

Reply:
Updated